File: AI/src/database/RoadMap_Schema.py
from mongoengine import Document, StringField, IntField, MapField, ListField, DateTimeField
from datetime import datetime

class RoadMap(Document):
    goal = StringField()
    level = StringField(required=False)
    description = StringField()
    estimated_hours = IntField()
    userId = StringField()
    createdAt = DateTimeField(default=datetime.now)
    roadmapId = StringField()

    skills = MapField(field=ListField(StringField()))

    meta = {
        'collection' : 'Road_Map',
        'ordering': ['-id']
    }

File: backend/src/modules/ai-learning-paths/schema/roadmap.schema.ts
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { HydratedDocument } from 'mongoose';

export type RoadMapDocument = HydratedDocument<RoadMap>;

@Schema({ timestamps: true, collection: 'Road_Map' })
export class RoadMap {
  @Prop({ type: String })
  goal: string;

  @Prop({ type: String, required: false })
  level?: string;

  @Prop({ type: String })
  description: string;

  @Prop({ type: Number })
  estimated_hours: number;

  @Prop({ type: String })
  userId: string;

  @Prop({ type: Date, default: Date.now })
  createdAt: Date;

  @Prop({ type: String })
  roadmapId: string;

  @Prop({ type: Object })
  skills: Record<string, string[]>;

  @Prop({ type: String, required: false })
  imageUrl?: string;
}

export const RoadMapSchema = SchemaFactory.createForClass(RoadMap);

